src/murrine_style.c

```diff
@@ -2161,6 +2160,11 @@ murrine_style_draw_layout (GtkStyle     *style,
 		{
 			boolean use_parentbg = TRUE;
 
+			if (state_type == GTK_STATE_PRELIGHT) {
+				use_parentbg = FALSE;
+				goto draw;
```

Comment:
It seems like you're really just using this `goto` to skip the `while` loop. Instead could you just do:

```
if (state_type == GTK_STATE_PRELIGHT) {
  use_parentbg = FALSE;
} else {
  while (widget->parent) {
     ...
  }
}
```

?